### PR TITLE
Node failure tests for cluster autoscaler

### PR DESCRIPTION
**Pull request by mwielgus**
E2e tests checking whether CA is still working with a single broken node.

cc: @MaciekPytel @jszczepkowski @fgrzadkowski

### CODE REVIEW

**Comment by mwielgus** [comment]
@k8s-bot verify test this

**Comment by MaciekPytel** [comment]
\lgtm

**Comment by mwielgus** [comment]
@k8s-bot verify test this

**Comment by mwielgus** [comment]
@k8s-bot gce etcd3 e2e test thi

**Comment by mwielgus** [comment]
@k8s-bot test this

File: test/e2e/cluster_size_autoscaling.go [hunk 1]
```diff
@@ -642,6 +667,21 @@ func waitForAllCaPodsReadyInNamespace(f *framework.Framework, c clientset.Interf
...
+func getAnyNode(c clientset.Interface) *v1.Node {
+	nodes, err := c.Core().Nodes().List(metav1.ListOptions{FieldSelector: fields.Set{
+		"spec.unschedulable": "false",
+	}.AsSelector().String()})
+	if err != nil {
+		glog.Error("Failed to get node list: %v", err)
+		return nil
+	}
+	if len(nodes.Items) == 0 {
+		glog.Errorf("No nodes")
```
Comment (jszczepkowski) [inline test/e2e/cluster_size_autoscaling.go]: s/Errorf/Error

File: test/e2e/cluster_size_autoscaling.go [hunk 2]
```diff
@@ -642,6 +667,21 @@ func waitForAllCaPodsReadyInNamespace(f *framework.Framework, c clientset.Interf
 	return fmt.Errorf("Some pods are still not running: %v", notready)
 }
 
+func getAnyNode(c clientset.Interface) *v1.Node {
+	nodes, err := c.Core().Nodes().List(metav1.ListOptions{FieldSelector: fields.Set{
+		"spec.unschedulable": "false",
+	}.AsSelector().String()})
+	if err != nil {
+		glog.Error("Failed to get node list: %v", err)
```
Comment (jszczepkowski) [inline test/e2e/cluster_size_autoscaling.go]: s/Error/Errorf

File: test/e2e/cluster_size_autoscaling.go [hunk 3]
```diff
@@ -291,12 +299,29 @@ var _ = framework.KubeDescribe("Cluster size autoscaling [Slow]", func() {
...
+
 		By("Some node should be removed")
 		framework.ExpectNoError(WaitForClusterSizeFunc(f.ClientSet,
 			func(size int) bool { return size < increasedSize }, scaleDownTimeout))
-	})
+	}
+
+	It("should correctly scale down after a node is not needed [Feature:ClusterSizeAutoscalingScaleDown]", simpleScaleDownTest)
+
+	It("should increase cluster size if pending pods are small and one node is broken [Feature:ClusterSizeAutoscalingScaleUp]",
```
Comment (jszczepkowski) [inline test/e2e/cluster_size_autoscaling.go]: s/increase/decrease

Comment (MaciekPytel) [inline test/e2e/cluster_size_autoscaling.go]: pretty sure increase is correct here (this is a scale up test)

**Pull request merged**